### ISSUE TITLE
GGC and GGMax Request: Combo Key Input

### ISSUE THREAD

**Issue by Teabone3** [TheGameCreators/GameGuruRepo#3670]
Currently there is no way to detect if a combination of keys are being pressed in Game Guru via Lua unless you are using the predefined keys which are limited.

For example there is no way to detect if the up-arrow key and the side arrow key are being pressed at the same time.

**Comment by Necrym59** [comment]
This is a problem within Lua, I have asked Lee for a virtual key alias input command eg: 'SendEmulatedKey(keyscancode)' or somthing similar hopefully Lee will look into this as it would solve not only your problem but the whole key and device button bindings etc in one fell swoop.

**Comment by Teabone3** [comment]
g_Scancode can only return 1 value and it detects the lowest value first. So if you were to hold 200 and 203 together (up and right) it would only detect the up key.

G_keyPress on the other hand just checks if each case is true and therefore you can have working combinations. Would be great to have g_KeyPressUP, g_KeyPressDOWN, g_KeyPressRIGHT, g_KeyPressLEFT. Similar to g_KeyPressSHIFT and g_KeyPressSPACE, however that would only open up the arrow keys for detection and not other possibilities beyond whats already listed in the g_KeyPress functions list.

**Comment by Necrym59** [comment]
You could try using
GetKeyState: x=GetKeyState(y) -- returns 1 in x when keycode specified in y is pressed
then detect if both = 1 then do whatever

**Comment by LeeBamberTGC** [comment]
@Necrym59 @Teabone3 Yes the command "GetKeyState" is available in both Classic and MAX.

**Comment by Teabone3** [comment]
GetKeyState has never seemed to work for me. It doesn't use ScanCode nor string input. So not sure what value sheet to refer to to use it. Says "keycode" but i dont have a reference list for that.

Also, before its mentioned, the command g_InKey that supports inputs as string, does not work for key combinations.

**Comment by Necrym59** [comment]
try a test with
Text(50,50,3,GetKeyState(200))
Text(50,53,3,GetKeyState(205))
I tried and they both return 1 each even when used simutaneously so i used this keystates map
![Keystate](https://user-images.githubusercontent.com/95015657/192137146-ca6b0728-8d96-4e0b-91af-e8d1321eccfb.png)

**Comment by LeeBamberTGC** [comment]
@Necrym59 Where did you get the graphic, it would be good to add to MAX docs folder for reference. 

@Teabone3 The keycode is a reference to what we call keyboard hardware code which differs from scancodes or ascii codes.  It is basically how DirectX mapped the keyboards way back when.

**Comment by Necrym59** [comment]
To be honest i cant remember i thinkl i got it from the game creators forum years ago in the Dark Basic days.  feel free to use it.

**Comment by Teabone3** [comment]
Thanks for this key map, Its kind of funny I think I used to carry this image on multiple computers over the years not sure what it was in reference to. I think i used to use it for FPI scripting with FPSC? 

This seems to have worked. I'm essentially trying to test combo keys to make sure the character also does NOT do actions when various keys are held in combo to those which cause an action, ironically. 

Noticed in GGC the 3rd person vanilla controls have some issues with 3+ key combos and end up making the character animate in ways his actions dont relate to - in the rare case a player does hold multiple keys. But i dont think too many people are using the 3rd person in GGC anymore -- so not a huge issue. This is more for custom 3rd person im working on, that should theatrically work both in classic and max though currently focused on classic as its for a game I'm working and hoping to complete some day.

Thanks for the keymap I'll see if it works and also get someone to test who uses a different language keyboard.
